luigi/hadoop.py

```diff
@@ -146,6 +168,16 @@ def add_files_for_package(sub_package_path, root_package_path, root_package_name
                 add(dereference(f), os.path.basename(f))
             else:
                 add(dereference(f), n + ".py")
+
+    # Add any egg-info directories that are needed by the modules
+    # and packages that are included in this tarball.
+    for egg_info_root_path in egg_info_to_add:
+        logger.debug(
+            'Adding package metadata to archive found at "%s"',
+            egg_info_root_path
+        )
+        add(egg_info_root_path, os.path.basename(egg_info_root_path))
```

Comment:
Is it weird that this call is going to pass in the egg_info_root_path to potentially call egg_info_to_add.add(egg_info_path)?  It probably shouldn't matter because it's implemented as a set(), but it still seems weird.  The way around is to have an add_to_tar() that is used here and in add().